### DISCUSSION TITLE
Marker wechseln/verschwinden lassen

### DISCUSSION THREAD

**Discussion by ue-black** [e-netze/datalinq-community#16] [General]
Hallo!

Ich habe folgende Karte:
![image](https://github.com/gis-eni/datalinq-forum/assets/127403099/78af577d-52db-4e1b-8001-481d2b58e9e9)

und würde die Marker gerne Verschwinden lassen bzw. mit einem anderen Marker als den marker_blue.png ersetzen. Kann ich das im Datalinq irgendwo einstellen?

Vielen Dank für die Info!

PS: Es sind nur Testdaten, wir haben in Wirklichkeit überall ausgezeichnete Badestellen 😄

**Comment by gis-eni** [comment]
Hallo!

Wenn du mit den DataLinqHelper Methoden eine Karte in DataLinq einfügst, werden auch automatisch die Marker angezeigt.
Ich verstehe den Sinn nicht, die Marker nicht anzuzeigen. Was würde die Karte dann anzeigen?

Was möglich wäre:
Die Marker sind "Leaflet" Marker und haben auch diverse Styles. Du kannst die beispielsweise mit den Developer Werkzeugen im Browser anzeigen lassen, was dahinter steht. Vielleicht kannst du für die Seite diese Styles so überschreiben, dass die Marker nicht angezeigt werden:

``` css
  ... { display: none }
```

Oder über Javascript ähnlich, wie das sonst in der ``custom.js`` gemacht werden kann.

Hast du in dieser Richtung schon was ausprobiert?

**Reply by gis-eni** [reply]
Konkret würde das heißen:

in der View:

Unsichtbar schalten von Markern und optional Popups für die Marker: 
``` html
<style>
    .leaflet-marker-icon, 
    .leaflet-popup {  
       display: none
    }
</style>
```

Ändern der Marker über Javascript (in einem ``<script>`` Block)

``` javascript
webgis.markerIcons["query_result"]["default"] = {
    url: function (i, f) {
        return webgis.baseUrl + '/rest/numbermarker/' + (i + 1);
    },
    size: function (i, f) { return [33, 41]; },
    anchor: function (i, f) { return [16, 42]; },
    popupAnchor: function (i, f) { return [0, -42]; },
    className: function (i, f) {
        return null; // "webgis-sprite-marker";
    }
};
```

**Comment by faulgurke** [comment]
Zum obigen Beispiel für die Änderung von Markern über Javascript habe ich noch eine Frage:
Wie müsste das query_result in webgis.markerIcons["query_result"] formuliert sein; oder noch besser ein konkretes Beispiel das funktionieren könnte (in JavaScript kenne ich mich einfach zu wenig aus).

**Reply by gis-eni** [reply]
Ist das oben kein konkretes Beispiel? Ich weiß leider nicht, was da bei euch an Daten ankommt. Aber mach einfach einmal ein 

``` javascript
console.log(f); 
``` 

in der **url function**. Dann siehst du was daher kommt.

**Reply by fanie-mak** [reply]
webgis.markerIcons["query_result"]["**_default oder ID DER ABFRAGE_**"] = 
  {
    url: function (i, f) 
    {
      

> //wenn man auf einen Attributwert der Abfrage zugreifen will - diese als Variable deklarieren und den Wert übergeben

      
      var myVariable = parseInt(f.properties.**_NAME_DES_ATTRIBUTES_IM_CMS_**, 10);
      

> //danach können Bereiche oder Kategorien definiert werden - das Bild liegt am Server

      if(myVariable >= 0 && myVariable < 10) return webgis.css.imgResource('point_white.png', 'markers');
      if (myVariable >= 10 && myVariable < 25) return webgis.css.imgResource('point_lightblue.png', 'markers');
      if (myVariable  >= 25 && myVariable < 50) return webgis.css.imgResource('point_blue.png', 'markers');
      if (myVariable  >= 50 && myVariable < 100) return webgis.css.imgResource('point_darkblue.png', 'markers') ;
      if (myVariable  >= 100) return webgis.css.imgResource('point_red.png', 'markers');  
      //return webgis.css.imgResource('point_blue.png', 'markers');                                                
    },
    size: function (i, f) { return [33, 41]; },  
    anchor: function (i, f) { return [11, 11]; },
    popupAnchor: function (i, f) { return [0, -11];},
    };

**Reply by faulgurke** [reply]
Vielen Dank Stephanie!! Das ist für jemanden wie mich zum Nachbauen.
